packages/block-library/src/navigation/edit.js

```diff
@@ -85,11 +84,11 @@ function Navigation( {
 		{
 			allowedBlocks: ALLOWED_BLOCKS,
 			orientation: attributes.orientation || 'horizontal',
-			renderAppender:
+			appender:
 				( isImmediateParentOfSelectedBlock &&
 					! selectedBlockHasDescendants ) ||
 				isSelected
-					? InnerBlocks.DefaultAppender
```

Comment:
If we update it to the correct `DefaultBlockAppender`, it doesn't get the expected behavior (only the plus button), so I updated it to `undefined`, keeping the same previous behavior.